verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -74,33 +77,27 @@ void SymbolTableHandler::LoadProjectFileList(absl::string_view current_dir) {
   LOG(INFO) << __FUNCTION__;
   if (!currproject) return;
   // search for FileList file up the directory hierarchy
-  std::filesystem::path currentdirpath{current_dir.begin(), current_dir.end()};
-  std::filesystem::path projectpath = currentdirpath / filelistname;
   FileList filelist;
-  while (true) {
-    projectpath = currentdirpath / filelistname;
-    // file found
-    LOG(INFO) << "Checking existence of " << projectpath;
-    if (std::filesystem::exists(projectpath)) break;
-    // file not found, reached root directory
-    if (currentdirpath.parent_path() == currentdirpath) {
-      LOG(INFO) << filelistname << " not found";
-      return;
-    }
-    currentdirpath = currentdirpath.parent_path();
+  std::string projectpath;
+  if (verible::file::UpwardFileSearch(current_dir,
+                                      absl::GetFlag(FLAGS_file_list_path),
+                                      &projectpath) != absl::OkStatus()) {
+    LOG(WARNING) << "Could not find " << absl::GetFlag(FLAGS_file_list_path)
+                 << " file in the project";
+    return;
   }
   LOG(INFO) << "Found file list under " << projectpath;
   // fill the FileList object
-  absl::Status status = AppendFileListFromFile(projectpath.string(), &filelist);
+  absl::Status status = AppendFileListFromFile(projectpath, &filelist);
```

Comment:
Similar here, if we include the message returned in the status, the log-message might be more useful

```c++
if (auto status = AppendFileListFromFile(...); !status.ok()) {
   LOG(WARNING) << "..." << status;
   return;
}